sabnzbd/nzbstuff.py

```diff
@@ -371,15 +373,72 @@ def __init__(self, date, subject, raw_article_db, file_bytes, nzo):
             # Any articles left?
             if raw_article_db:
                 # Save the rest
-                sabnzbd.save_data(raw_article_db, self.nzf_id, nzo.admin_path)
+                self.finish_import(raw_article_db)
+                self.pickle_articles(initial_pickle=True)
             else:
                 # All imported
                 self.import_finished = True
 
-    def finish_import(self):
-        """ Load the article objects from disk """
+    def pickle_articles(self, initial_pickle=False):
+        """ Pickle articles to file """
+        if not initial_pickle:
+            if (
+                not self.import_finished
+                # Don't pickle tiny files
+                or len(self.articles) < 3
+                # Don't pickle active files
+                or self.pickle_lock_time > time.time()
```

Comment:
You will need to set `self.pickle_lock_time = 0` in `__getstate__`, otherwise jobs from before this change will fail as non-existing attributes are set to `None`.
```python
>>> None > 1
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

Reply:
Thanks. this lead me to find a couple of other bugs. The `TypeError` bug would actually have been triggered very rarely because I didn't initialize `nzo.unpickled_files` either, so it would only have tried to pickle them in very special circumstances. Now an existing unpickled queue should be pickled after an upgrade if pickling is enabled.